core/modules/layout/layout.flexible.inc

```diff
@@ -488,24 +488,43 @@ function layout_flexible_template_edit_row_form($form, &$form_state, LayoutFlexi
 
   $form['region_names'] = array(
     '#type' => 'fieldset',
-    '#title' => t('Region names'),
+    '#title' => t('Regions'),
     '#collapsed' => $original_row != 'add',
     '#collapsible' => TRUE,
     '#tree' => TRUE,
   );
 
   $form_state['last_region_number'] = layout_flexible_template_get_last_region_number($flexible_template);
   for ($i = 0; $i < $region_count; $i++) {
-    $form['region_names']['region_name_' . $i] = array(
+    $form['region_names']['region_' . $i] = array(
+      '#type' => 'container',
+    );
+    $form['region_names']['region_' . $i]['region_name'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Region name') . ' ' . ($i + 1),
+      '#default_value' => !empty($row_data['region_names']['region_' . $i]['region_name']) ? $row_data['region_names']['region_' . $i]['region_name'] : t('Region') . ' ' . ($form_state['last_region_number'] + $i + 1),
```

Comment:
This line gets extremely long, which is bad for readability. I wonder if it makes sense to do a regular if/else instead.

Reply:
@indigoxela I've changed this. Interested in doing a review?